eng/DownloadSharedSource.ps1

```diff
@@ -2,29 +2,89 @@
 $ErrorActionPreference = 'Stop'
 $ProgressPreference = 'SilentlyContinue'
 
-function DownloadAll([string[]]$files, [string]$baseUrl, [string]$downloadPath)
-{
-    foreach ($file in $files)
-    {
-        Write-Host "Downloading" $file
-        $text = (Invoke-WebRequest -Uri "$baseUrl/$file").Content
-        $text.Trim() | Out-File (Join-Path $downloadPath $file)
+$sparseCheckoutFile = ".git/info/sparse-checkout"
+$gitRemoteUrl = "https://github.com/Azure/azure-sdk-for-net.git"
+$gitRemoteBranchName = "main"
+$azCoreSharedPath = "sdk/core/Azure.Core/src/Shared/"
+$armCoreSharedPath = "sdk/resourcemanager/Azure.ResourceManager/src/Shared/"
+
+function InitializeSparseGitClone([string]$repo) {
+    git clone --no-checkout --filter=tree:0 $repo .
+    if ($LASTEXITCODE) { exit $LASTEXITCODE }
+    git sparse-checkout init
+    if ($LASTEXITCODE) { exit $LASTEXITCODE }
+    Remove-Item $sparseCheckoutFile -Force
+}
+
+function AddSparseCheckoutPath([string]$subDirectory) {
+    if (!(Test-Path $sparseCheckoutFile) -or !((Get-Content $sparseCheckoutFile).Contains($subDirectory))) {
+        Write-Output $subDirectory >> $sparseCheckoutFile
+    }
+}
+
+function GetSDKCloneDir() {
+    $root = git rev-parse --show-toplevel
+
+    $sparseSpecCloneDir = "$root/sdk"
+    New-Item $sparseSpecCloneDir -Type Directory -Force | Out-Null
+    $createResult = Resolve-Path $sparseSpecCloneDir
+    return $createResult
+}
+
+$sdkCloneDir = GetSDKCloneDir
+Write-Host "Setting up sparse clone for $gitRemoteUrl at $sdkCloneDir"
+
+Push-Location $sdkCloneDir.Path
+try {
+    if (!(Test-Path ".git")) {
+        Write-Host "Initializing sparse clone for $gitRemoteUrl"
+        InitializeSparseGitClone $gitRemoteUrl
+        AddSparseCheckoutPath $azCoreSharedPath
+        AddSparseCheckoutPath $armCoreSharedPath
+        $commitHash = git rev-parse $gitRemoteBranchName
+        git checkout $commitHash
+    }
+    if ($LASTEXITCODE) { exit $LASTEXITCODE }
+}
+finally {
+    Pop-Location
+}
+
+function CopyFilesToDestination([string]$sdkCloneRoot, [string]$dirName, [string]$dest, [string[]]$files) {
+    $source = Join-Path $sdkCloneRoot $dirName
+    foreach ($file in $files) {
+        $sourceFile = Join-Path $source $file
+        $destFile = Join-Path $dest $file
+        # to avoid the format issues (the file is required to end by a newline)
```

Comment:
I'm not sure I understand this issue. If the newlines don't match then we should fix this in the .NET repo.

Reply:
I fixed this in the .NET repo:
https://github.com/Azure/azure-sdk-for-net/pull/36250

Reply:
There's no standard encoding or trailing new-line rule in the .NET repo but this repo's build enforces trailing new lines.  We should either relax that rule here or start enforcing the rule for shared code in azure-sdk-for-net

Reply:
Then let me change it back so that those unwanted changes will disappear as well.